python/cuspatial/cuspatial/tests/test_traj.py

```diff
@@ -327,3 +328,29 @@ def test_distance_and_speed_single_trajectory():
         cudf.Series([1973230.625, 2270853.0, 4242640.5]),
         check_names=False,
     )  # fast!
+
+
+@pytest.mark.parametrize(
+    "timestamp_type",
+    [
+        ("datetime64[ns]", 1000000000),
+        ("datetime64[us]", 1000000),
+        ("datetime64[ms]", 1000),
+        ("datetime64[s]", 1),
+    ],
+)
+def test_distance_and_speed_timestamp_types(timestamp_type):
+    result = cuspatial.distance_and_speed(
+        cudf.Series([0.0, 0.001, 0.0, 0.0]),
+        cudf.Series([0.0, 0.0, 0.0, 0.001]),
+        cudf.Series([0, timestamp_type[1], 0, timestamp_type[1]]).astype(
+            timestamp_type[0]
+        ),
+        cudf.Series([2, 2]),
+        cudf.Series([2, 4]),
+    )
+    assert_eq(
+        result,
+        cudf.DataFrame({"meters": [1.0, 1.0], "speed": [1.0, 1.0]}),
+        check_names=False,
```

Comment:
The timestamps are always cast to `ms` on their way in. I *think* what we are seeing here is a multiplier of 1000 to convert from `ms` to `s`, and another multiplier of `1000` to convert from `km/s` to `m/s`.

Reply:
Maybe we should have the return distance scale be set to `km` @zhangjianting , since `km` is the input scale? In that case we'd just return `ds` for `distance` and `ds * 1000 / unwrap(td)` for time - or we force the timestamp to `ns` instead of `ms` for better precision.

Reply:
OK, so the fact that the kernel assumes the timestamp in ms does happen to be correct for cuDF behavior?  I assume that m/s is a more intuitive output than km/s. If you change to km you would need to go to km/h. I think the input should probably be in m to match the output. But maybe this is a geospatial standard. Will let @zhangjianting advise.